src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -427,9 +427,12 @@ internal bool TryUnparse(StringBuilder sb)
         public class Options
         {
             /// <summary>
-            /// Whether the input may include quoted values, which can contain separator characters, colons,
```

Comment:
>colons [](start = 101, length = 6)

Does this argument still work for colons? If yes, then this should be mentioned.

Reply:
@yaeldekel do you mean specifically colons in sparse datasets or generally? 

Sparse data with quotes, e.g. the following, is read as non-sparse regardless of the `allowQuoting` setting.
C1	C2	C3	C4	C5
1	2	3	4	5
5	"3:4"	4:5
5	1:2

The third line is read as
5	?	0	0	0

Generally speaking, any special characters within quotes are read correctly with `allowSparse = true`. e.g. the second line in this file: https://github.com/dotnet/machinelearning/blob/master/test/data/QuotingData.csv#L3
2,"This text contains extra punctuations and special characters.;*<>?!@#$%^&*()_+=-{}|[]:;'"

is read as:
2	"This text contains extra punctuations and special characters.;*<>?!@#$%^&*()_+=-{}|[]:;'"

Do you have an opinion on how this should be documented?

Reply:
What I was referring to was the case where there is a text column, and the text may contain a colon. What would be the combination of values for `AllowQuoting` and `AllowSparse` that would make TextLoader parse that correctly? I think that if `AllowSparse` is true, then you need `AllowQuoting` to be true, **and** the text containing the colon has to be in quotes (but don't take my word for it, I'm not sure if this is the case), and if `AllowSparse` is false, then the colon will be read as part of the text in the column whether there are quotes or not.

---
In reply to: [387406874](https://github.com/dotnet/machinelearning/pull/4899#discussion_r387406874) [](ancestors = 387406874)

Reply:
I don't think that these special characters need any quotes for TextLoader to be able to read them, the only characters are treated differently based on the value of `AllowQuoting` are the separator characters, and colon (since it has a special meaning in the sparse case).

---
In reply to: [387531140](https://github.com/dotnet/machinelearning/pull/4899#discussion_r387531140) [](ancestors = 387531140,387406874)

Reply:
I'll verify this and update accordingly.

Reply:
@yaeldekel the treatment of colons is independent of `allowQuoting` for both sparse and dense data.

This dense row containing a colon is read correctly regardless of `allowQuoting` and regardless of whether the second column is within quotes or not.
`2,This text contains extra punctuations and special characters.;*<>?!@#$%^&*()_+=-{}|[]:;'`

This sparse row is read correctly (when read as `DataKind.String`) with `allowSparse = true`, regardless of `allowQuoting` and regardless of whether the value `"4:5"` is within quotes or not. 
`5	3:"4:5"	4:5`

The only difference with `allowQuoting = true` is that the value is read as `4:5` instead of `"4:5"`, which is the same as the behavior for any other kind of text, not specific to colons.

I'm going to leave this as it is. #Resolved